connectors/sources/sharepoint.py

```diff
@@ -302,7 +166,6 @@ async def _generate_session(self):
 
         if self.is_cloud:
             basic_auth = None
-            await self._set_access_token()
```

Comment:
We have kept these two things different because we are going to check `_get_session` for every api call and if we include access token here then this will also get check every time.

Reply:
I have kept  `_set_access_token` and  `get_session`  separate because for every `_api_call` we have to check `session` and `access_token`